supabase/tests/draft_collections_eligible_for_deletion.test.sql

```diff
@@ -0,0 +1,147 @@
+create function tests.test_draft_collections_eligible_for_deletion()
+returns setof text as $$
+declare
+  target_capture_id flowid;
+  target_draft_id flowid;
+
+  collection_a_id flowid;
+  collection_b_id flowid;
+  collection_c_id flowid;
+  collection_d_id flowid;
+  collection_e_id flowid;
+
+  capture_consumer_id flowid;
+  materialization_consumer_id flowid;
+  derivation_consumer_c_id flowid;
+  derivation_consumer_d_id flowid;
+begin
+
+  insert into user_grants (user_id, object_role, capability) values
+    ('33333333-3333-3333-3333-333333333333', 'carolCo/', 'admin');
```

Comment:
I think this test setup should include a materialization/derivation that the user doesn't have access to and reads from a collection that would otherwise be eligible for deletion. This would help validate that the RLS policy changes are all working right.

Reply:
I think this still could use an assertion that does not populate the draft with specs you don't have access to, for example when calling `draft_collections_eligible_for_deletion` with the id of a capture belonging to another user. It seems like if you did:

```plpgsql
perform set_authenticated_context('22222222-2222-2222-2222-222222222222');
perform draft_collections_eligible_for_deletion(target_capture_id, target_draft_id);
```

then it should either return an error or result in no specs added to the draft. Except that `draft_collections_eligible_for_deletion` still has `security definer`, so I'd think it might actually push the `carolCo` specs into the `bobCo` draft.